metals/src/main/scala/scala/meta/internal/metals/BatchedFunction.scala

```diff
@@ -90,9 +113,17 @@ final class BatchedFunction[A, B](
     try {
       if (requests.nonEmpty) {
         val args = requests.flatMap(_.arguments)
+        val callbacks = requests.map(_.callback)
         val result = fn(args)
         this.current.set(result)
-        result.future.onComplete { response =>
+        val resultF = for {
+          result <- result.future
+          _ <- Future {
+            val callbackData = BatchedCallbackData(args, result)
```

Comment:
Do we need `BatchedCallbackData` at all? This doesn't seem to be used

Reply:
I wanted to create a proper callback, may be I should have used `Unit => Unit`